internal/services/rate.go

```diff
@@ -23,15 +23,15 @@ type Rate struct {
 
 var ErrNoFetchOccurred = errors.New("no fecth occurred yet")
 
-type RateService struct {
+type CachingRateService struct {
```

Comment:
'Caching' prefix is redundant. Other code have just use this service without knowing that it has caching mechanism

Reply:
Same goes here, this is a concrete implementation. Handlers don't know too much about details and just use interface `service.RateService`